### PR TITLE
[AMORO-2078] Replace set with bitmap when apply pos delete records

### LINKED ISSUE #2078: [Improvement]: Change position delete index from set to bitmap of optimizing
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, the optimization task of Iceberg uses a Set data structure for indexing position-delete data, which consumes a lot of memory. It could be changed to a bitmap to reduce memory usage.

### How should we improve?

Modify com.netease.arctic.iceberg.CombinedDeleteFilter position delete index

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2078.

## Brief change log

- Reduce memory for large pos-deletes

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -119,6 +119,12 @@
             <artifactId>paimon-bundle</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.roaringbitmap</groupId>
+            <artifactId>RoaringBitmap</artifactId>
+            <version>0.9.25</version>
```
Comment (zhoujinsong) [inline core/pom.xml:125]: I recommend managing the dependency in the `DependencyManagement` blocker in parent pom and store the version in properties.

Comment (zhoujinsong) [reply core/pom.xml:125]: BTW, should we dependent on version 0.9.44 as Iceberg project do.

Comment (shidayang) [inline core/pom.xml:125]: Add dependency to dependencyManager

**Review by shidayang — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/netease/arctic/io/reader/CombinedDeleteFilter.java [hunk 2]
```diff
@@ -79,7 +80,7 @@ public abstract class CombinedDeleteFilter<T extends StructLike> {
   private final List<DeleteFile> posDeletes;
   private final List<DeleteFile> eqDeletes;
 
-  private Map<String, Set<Long>> positionMap;
+  private Map<String, Roaring64Bitmap> positionMap;
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/io/reader/CombinedDeleteFilter.java:83]: Maybe we can resue the `org.apache.iceberg.deletesPositionDeleteIndex` here?

**Review by zhoujinsong — COMMENTED** [review]
@XBaith I left some comments.

File: core/src/main/java/com/netease/arctic/io/reader/CombinedDeleteFilter.java (hunk 2 above)
Comment (XBaith) [reply core/src/main/java/com/netease/arctic/io/reader/CombinedDeleteFilter.java:83]: `Deletes.positionDeleteIndex` reqire datafile path which is not exist in `CombinedDeleteFilter`

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
LGTM

**Pull request merged**